### PR TITLE
Patch CVE-2023-44487 in vendored golang

### PR DESCRIPTION
**Pull request by dmcilvaney**
Also unify how we handle vendored patching across
the affected golang packages.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Resolve CVE-2023-44487 aka CVE-2023-39325 (for golang specifically) by patching the vulnerable vendored go sources we use.

###### Change Log  

- Resolve CVE-2023-44487 aka CVE-2023-39325 in:
  - application-gateway-kubernetes-ingress
  - azcopy
  - cf-cli
  - containerized-data-importer
  - coredns
  - git-lfs
  - jx
  - keda
  - kube-vip-cloud-provider
  - kubevirt
  - local-path-provisioner
  - moby-buildx
  - moby-compose
  - moby-containerd
  - moby-containerd-cc
  - packer
  - prometheus-node-exporter
  - prometheus-process-exporter
  - rook
  - terraform
  - vitess

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/48735735

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-44487
- https://nvd.nist.gov/vuln/detail/CVE-2023-39325

###### Test Methodology

- Local builds, PR gates

### CODE REVIEW

File: SPECS/cf-cli/cf-cli.spec [hunk 1]
```diff
@@ -27,19 +27,26 @@ Source0:        https://github.com/cloudfoundry/cli/archive/refs/tags/v%{version
 #         See: https://reproducible-builds.org/docs/archives/
 #       - For the value of "--mtime" use the date "2021-04-26 00:00Z" to simplify future updates.
 Source1:        cli-%{version}-vendor.tar.gz
+Patch0:         use-mod-vendor-with-tests.patch
+Patch1:         CVE-2023-44487.patch
```
Comment (dmcilvaney) [inline SPECS/cf-cli/cf-cli.spec:31]: Not needed without new check section...
```suggestion
Patch0:         CVE-2023-44487.patch
```

File: SPECS/jx/jx.spec [hunk 2]
```diff
@@ -60,6 +65,9 @@ install -p -m 755 -t %{buildroot}%{_bindir} ./build/jx
 %{_bindir}/jx
 
 %changelog
+* Thu Feb 01 2024 Daniel McIlvaney <damcilva@microsoft.com> -3.2.236-15
+- Address CVE-2023-44487 by patching vendored golang.org/x/net
```
Comment (dmcilvaney) [inline SPECS/jx/jx.spec:69]:
```suggestion
- Address CVE-2023-44487 by patching vendored golang.org/x/net
- Add unit tests to check section
```

File: SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec [hunk 3]
```diff
@@ -23,28 +23,36 @@ Source0:        https://github.com/kube-vip/%{name}/archive/refs/tags/v%{version
...
+%check
+go test -mod=vendor ./...
+
 %files
 %{_bindir}/kube-vip-cloud-provider
 
 %changelog
+* Wed Feb 07 2024 Daniel McIlvaney <damcilva@microsoft.com> - 0.0.2-16
+- Address CVE-2023-44487 by patching vendored golang.org/x/net
+- Rework CVE-2023-21698.patch to apply without directory change
```
Comment (dmcilvaney) [inline SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec:54]:
```suggestion
- Rework CVE-2023-21698.patch to apply without directory change
- Add check section
```

File: SPECS/rook/rook.spec [hunk 4]
```diff
@@ -126,8 +127,10 @@ This package contains Helm Charts for Rook.
 
 %prep
 %autosetup -N
+# Apply vendor before patching
 tar -xf %{SOURCE1} --no-same-owner
 %autopatch -p1
+%autopatch -p1
```
Comment (dmcilvaney) [inline SPECS/rook/rook.spec:133]: Bad rebase
```suggestion
```

**Review by nicogbg — APPROVED** [review]

File: SPECS/vitess/vitess.spec [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 Name:           vitess
 Version:        16.0.2
-Release:        6%{?dist}
+Release:        7%{?dist}
```
Comment (dmcilvaney) [inline SPECS/vitess/vitess.spec:6]: %check section is failing, `go check` isn't a real command. Likely unrelated to these changes.

**Comment by dmcilvaney** [comment]
`promethus-process-exporter` has a known failure in %check, unchanged.

File: SPECS/jx/jx.spec [hunk 6]
```diff
@@ -36,10 +37,12 @@ BuildRequires:  golang >= 1.17.1
 Command line tool for working with Jenkins X.
 
 %prep
-%autosetup -p1
+%autosetup -N -p1
```
Comment (tobiasb-ms) [inline SPECS/jx/jx.spec:40]: nit: With the `-N` option, it won't apply patches, which renders `-p1` moot.
```suggestion
%autosetup -N
```

File: SPECS/keda/keda.spec [hunk 7]
```diff
@@ -29,16 +29,20 @@ Source1:        %{name}-%{version}-vendor-v2.tar.gz
 # Patches the version of client_golang used in the vendored source. Should be applied before creating the vendored tarball.
 # Can be removed if we upgrade keda to 2.6.0 or later.
 Patch0:         CVE-2022-21698.patch
+# Patches for vendored code >= 1000
```
[the remark this answers is not in the document]
Comment (tobiasb-ms) [reply SPECS/keda/keda.spec:32]:
```suggestion
```

**Pull request merged**